apps/game_backend/lib/game_backend/users/currencies.ex

```diff
@@ -86,10 +100,41 @@ defmodule GameBackend.Users.Currencies do
   end
 
   @doc """
-  Returns a boolean indicating whether the user can afford the required amount of the specified currency.
+  Returns whether the user can afford the required amounts of the specified currencies.
+  """
+  def can_afford(user_id, currencies_list) do
+    Enum.all?(currencies_list, fn %CurrencyCost{currency_id: currency_id, amount: amount} ->
+      can_afford(user_id, currency_id, amount)
+    end)
+  end
+
+  @doc """
+  Returns whether the user can afford the required amount of the specified currency.
   """
   def can_afford(user_id, currency_id, required_amount) do
     user_balance = get_amount_of_currency(user_id, currency_id)
     user_balance >= required_amount
   end
+
+  @doc """
+  Substracts all CurrencyCosts from the user.
+  Returns {:ok, results} or {:error, "failed"} tuples so it can be used on transactions.
+  """
+  def substract_currencies(_user_id, []), do: {:ok, []}
+
+  def substract_currencies(user_id, costs) do
+    result =
+      Enum.map(costs, fn %CurrencyCost{currency_id: currency_id, amount: cost} ->
+        add_currency(user_id, currency_id, -cost)
```

Comment:
If this fails in the middle of the map execution you may substract a currency and do not give the user the thing they wanted, not mandatory but you could add al the substractions to a multi with reduce instead of mapping the results

Reply:
Right, the idea is for this to be called inside a multi like you say (check `Units.level_up`). Do you think we should wrap this in a multi too? We'd have a multi inside a multi, I think that's fine right?

Reply:
But there you're missing the Repo.transaction() call, making the `substract_currencies` function do nothing on its own... What do you think about just clarifying that this can be used inside a multi to avoid this issue in the function doc instead?